rest/server_context.go

```diff
@@ -1200,21 +1203,112 @@ var tempConnectionDetailsForManagementEndpoints = func() (serverAddress string,
 	return base.UnitTestUrl(), base.TestClusterUsername(), base.TestClusterPassword(), "", "", ""
 }
 
-func (sc *ServerContext) ObtainManagementEndpoints() ([]string, error) {
+func (sc *ServerContext) ObtainManagementEndpointsAndHTTPClient() ([]string, *http.Client, error) {
 	clusterAddress, clusterUser, clusterPass, certPath, keyPath, caCertPath := tempConnectionDetailsForManagementEndpoints()
 	agent, err := initClusterAgent(clusterAddress, clusterUser, clusterPass, certPath, keyPath, caCertPath, sc.config.ServerReadTimeout)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	managementEndpoints := agent.MgmtEps()
 
+	httpClient := &http.Client{
+		Transport: agent.HTTPClient().Transport,
+	}
+
 	err = agent.Close()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	return managementEndpoints, httpClient, nil
+}
+
+func (sc *ServerContext) CheckPermissions(httpClient *http.Client, managementEndpoints []string, username, password string, requestedPermissions []string) (statusCode int, permissionResults map[string]bool, err error) {
```

Comment:
Can we rename this? And probably include a comment for more detail. It does a poor job at conveying what the function is doing.

"Check" is fairly ambiguous:
- Is it checking there are ANY permissions present?
- Is it checking that the user has ALL of the requested permissions?

I think the answer to both of those is no. But I had to read the code to figure that out :)

Reply:
Yeah fair do's will add a comment and think of a name

Reply:
Couldn't think of a better name for now but has a fairly detailed comment.